### PR TITLE
[SYCL] Add deprecation warnings for unstable keys of SYCL_DEVICE_ALLOWLIST

### PR DESCRIPTION
**Pull request by dm-vodopyanov**
This patch adds deprecation warnings for DeviceName and PlatformName
keys of SYCL_DEVICE_ALLOWLIST and suggests using stable BackendName,
DeviceType and DeviceVendorId keys instead of them

### CODE REVIEW

File: sycl/source/detail/allowlist.cpp [hunk 1]
```diff
@@ -241,6 +254,27 @@ AllowListParsedT parseAllowList(const std::string &AllowListRaw) {
...
+    std::cerr << "\nWARNING: " << DeprecatedKeyNameDeviceName << " and "
+              << DeprecatedKeyNamePlatformName
+              << " in SYCL_DEVICE_ALLOWLIST are deprecated. ";
+  } else if (IsDeprecatedKeyNameDeviceNameWasUsed) {
+    std::cerr << "\nWARNING: " << DeprecatedKeyNameDeviceName
+              << " in SYCL_DEVICE_ALLOWLIST is deprecated. ";
+  } else if (IsDeprecatedKeyNamePlatformNameWasUsed) {
+    std::cerr << "\nWARNING: " << DeprecatedKeyNamePlatformName
+              << " in SYCL_DEVICE_ALLOWLIST is deprecated. ";
+  }
```
Comment (vladimirlaz) [inline sycl/source/detail/allowlist.cpp:268]: NIT: Isn't it better to have two independent warnings instead of having a combined one?

**Review by vladimirlaz — DISMISSED** [review]

File: sycl/source/detail/allowlist.cpp [hunk 2]
```diff
@@ -241,6 +254,27 @@ AllowListParsedT parseAllowList(const std::string &AllowListRaw) {
...
+  if (IsDeprecatedKeyNameDeviceNameWasUsed &&
+      IsDeprecatedKeyNamePlatformNameWasUsed) {
+    std::cerr << "\nWARNING: " << DeprecatedKeyNameDeviceName << " and "
+              << DeprecatedKeyNamePlatformName
+              << " in SYCL_DEVICE_ALLOWLIST are deprecated. ";
+  } else if (IsDeprecatedKeyNameDeviceNameWasUsed) {
+    std::cerr << "\nWARNING: " << DeprecatedKeyNameDeviceName
+              << " in SYCL_DEVICE_ALLOWLIST is deprecated. ";
+  } else if (IsDeprecatedKeyNamePlatformNameWasUsed) {
+    std::cerr << "\nWARNING: " << DeprecatedKeyNamePlatformName
```
Comment (bader) [inline sycl/source/detail/allowlist.cpp:266]: Are you sure `std::cerr` is the right stream for a warning message? `std::cerr` is intended for error output.

File: sycl/source/detail/allowlist.cpp (hunk 1 above)
Comment (dm-vodopyanov) [reply sycl/source/detail/allowlist.cpp:268]: I thought the less warnings user see the better

File: sycl/source/detail/allowlist.cpp (hunk 2 above)
Comment (dm-vodopyanov) [reply sycl/source/detail/allowlist.cpp]: Sounds reasonable, fixed in https://github.com/intel/llvm/pull/4455/commits/d56f0fcf997515fb5c2ea5556e9c3edcc9edda86

**Review by bader — APPROVED** [review]

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**